sentry-okhttp/src/main/java/io/sentry/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -81,6 +81,10 @@ public open class SentryOkHttpInterceptor(
             val parentSpan = if (Platform.isAndroid()) scopes.transaction else scopes.span
             span = parentSpan?.startChild("http.client", "$method $url")
         }
+
+        // interceptors may change the request details, so let's update it here
+        // this only works correctly if SentryOkHttpInterceptor is the last one in the chain
+        okHttpEvent?.setRequest(request)
```

Comment:
probably better to have it in the `finally` block, just before the `finishSpan` call, for consistency.
In the meantime we change the request (we add our headers) and at the end we call the `beforeSpan` on the updated request, so it's better to set the updated request into the event?
Not a strong opinion, though